File: selectionGeometryPlus.js
import { SelectionGeometry } from "../../../core/bokeh_events";

let SelectionGeometryPlus = class SelectionGeometryPlus extends SelectionGeometry {
        constructor(geometry, matrixType, final) {
            super();
            this.geometry = geometry;
            this.matrixType = matrixType;
            this.final = final;
        }
        _to_json() {
            const { geometry, matrixType, final } = this;
            return Object.assign(Object.assign({}, super._to_json()), { geometry, final });
        }
};
SelectionGeometryPlus.__name__ = "SelectionGeometryPlus";
SelectionGeometryPlus = __decorate([
    event("selectiongeometryplus")
], SelectionGeometryPlus);
export { SelectionGeometryPlus };

File: selectionGeometryPlus.py
#!/usr/bin/env python3

from bokeh.events import SelectionGeometry

class SelectionGeometryPlus(SelectionGeometry):
    """ Essentially the same as SelectionGeometry, but also contains
        an extra String field 
    """
    event_name = 'selectiongeometryplus'

    def __init__(self, model, geometry=None, plus=None, final=True):
        self.geometry = geometry
        self.final = final
        self.plus = plus
        super().__init__(model=model)
